zerver/models/custom_profile_fields.py

```diff
@@ -106,21 +106,27 @@ class CustomProfileField(models.Model):
 
     FIELD_TYPE_DATA: List[FieldElement] = [
         # Type, display name, validator, converter, keyword
-        (SHORT_TEXT, gettext_lazy("Short text"), check_short_string, str, "SHORT_TEXT"),
-        (LONG_TEXT, gettext_lazy("Long text"), check_long_string, str, "LONG_TEXT"),
-        (DATE, gettext_lazy("Date picker"), check_date, str, "DATE"),
-        (URL, gettext_lazy("Link"), check_url, str, "URL"),
+        (DATE, gettext_lazy("Date"), check_date, str, "DATE"),
         (
             EXTERNAL_ACCOUNT,
             gettext_lazy("External account"),
             check_short_string,
             str,
             "EXTERNAL_ACCOUNT",
         ),
+        (URL, gettext_lazy("Link"), check_url, str, "URL"),
         (PRONOUNS, gettext_lazy("Pronouns"), check_short_string, str, "PRONOUNS"),
+        (LONG_TEXT, gettext_lazy("Text (long)"), check_long_string, str, "LONG_TEXT"),
+        (SHORT_TEXT, gettext_lazy("Text (short)"), check_short_string, str, "SHORT_TEXT"),
     ]
 
-    ALL_FIELD_TYPES = [*FIELD_TYPE_DATA, *SELECT_FIELD_TYPE_DATA, *USER_FIELD_TYPE_DATA]
+    ALL_FIELD_TYPES = [
+        FIELD_TYPE_DATA[0],
+        FIELD_TYPE_DATA[1],
+        *SELECT_FIELD_TYPE_DATA,
+        *FIELD_TYPE_DATA[2:],
```

Comment:
This seems difficult to read. I guess a better solution would be to not sort here, but instead do it in either `events.py` where we set `custom_profile_field_types` to pass the data to client or let the client handle the sorting order. @timabbott thoughts? We can do it in the client itself if we do not care about the order of field types in `register` response.

In either case, we can probably have two commits - one for renaming and one for reordering.

Reply:
I'm not sure what solution is best but agree this feels pretty inscrutable. 

Agreed with writing this as two commits.

Reply:
Yes, this seems better.

Reply:
Hello Tim and Sahil, thank you for reviewing this. I have made a mistake on the version control of this pull request. I am going to close this pull request and make a new one that addresses your feedback here.